src/decoder.rs

```diff
@@ -120,6 +123,33 @@ pub fn strip_junk_header(slice: &[u8]) -> io::Result<&[u8]> {
     Ok(&slice[slice.len()..])
 }
 
+/// Decodes range mappping bitfield string into index
+fn decode_rmi(rmi_str: &str, val: &mut BitVec<u8, Lsb0>) -> Result<()> {
+    val.clear();
+    if val.capacity() < rmi_str.len() * 6 {
+        val.reserve(rmi_str.len() * 6 - val.capacity());
+    }
+    unsafe {
+        // Safety: we just reserved enough space
+        val.set_len(rmi_str.len() * 6);
+    };
+
+    for (idx, &byte) in rmi_str.as_bytes().iter().enumerate() {
+        let byte = match byte {
+            b'A'..=b'Z' => byte - b'A',
+            b'a'..=b'z' => byte - b'a' + 26,
+            b'0'..=b'9' => byte - b'0' + 52,
+            b'+' => 62,
+            b'/' => 63,
+            _ => unreachable!("invalid rmi"),
```

Comment:
the function already returns a `Result`, so you should return a proper error here.
I assume these strings are "untrusted input data", and panic-ing in that case is not a good idea.